fdbserver/storageserver.actor.cpp

```diff
@@ -1176,13 +1176,20 @@ ACTOR Future<Version> waitForVersionActor(StorageServer* data, Version version,
 // If the latest commit version that mutated the shard(s) being served by the specified storage
 // server is below the client specified read version then do a read at the latest commit version
 // of the storage server.
-Version getRealReadVersion(VersionVector& ssLatestCommitVersions, Tag& tag, Version specifiedReadVersion)
-{
-	Version realReadVersion = ssLatestCommitVersions.hasVersion(tag) ? ssLatestCommitVersions.getVersion(tag) : specifiedReadVersion;
+Version getRealReadVersion(VersionVector& ssLatestCommitVersions, Tag& tag, Version specifiedReadVersion) {
+	Version realReadVersion =
+	    ssLatestCommitVersions.hasVersion(tag) ? ssLatestCommitVersions.getVersion(tag) : specifiedReadVersion;
 	ASSERT(realReadVersion <= specifiedReadVersion);
 	return realReadVersion;
 }
 
+// Find the latest commit version of the given tag.
+Version getLatestCommitVersion(VersionVector& ssLatestCommitVersions, Tag& tag) {
```

Comment:
Can this be a member of `VersionVector`?

Reply:
VersionVector already has a method called "getVersion()', which we are using here, but that method assumes that the given tag has an entry in the VersionVector. Shall I add a new method to VersionVector that doesn't make such an assumption?

Reply:
The method in VersionVector asserts if the tag is not in the vector, so the caller must check if the tag is in the vector or not? A method that returned a status rather than asserting might be a nice way to reduce caller burden, for cases where its unknown if the tag is in the vector or not.